src/formula/variant.cpp

```diff
@@ -841,97 +634,49 @@ bool variant::operator>(const variant& v) const
 
 variant variant::list_elements_add(const variant& v) const
 {
-	must_be(TYPE_LIST);
-	v.must_be(TYPE_LIST);
-
-	if( num_elements() != v.num_elements() )
-		throw type_error("Operator '.+' requires two lists of the same length");
-
-	std::vector< variant > res;
-	res.reserve(num_elements());
-
-	for(size_t i = 0; i < num_elements(); ++i) {
-		res.push_back( (*this)[i] + v[i] );
-	}
-
-	return variant( &res );
+	must_both_be(VARIANT_TYPE::TYPE_LIST, v);
+	return value_cast<game_logic::variant_list>()->list_op(v.value_, [this](variant& v1, variant& v2) { return v1 + v2; });
 }
 
 variant variant::list_elements_sub(const variant& v) const
 {
-	must_be(TYPE_LIST);
-	v.must_be(TYPE_LIST);
-
-	if( num_elements() != v.num_elements() )
-		throw type_error("Operator '.-' requires two lists of the same length");
-
-	std::vector< variant > res;
-	res.reserve(num_elements());
-
-	for(size_t i = 0; i < num_elements(); ++i) {
-		res.push_back( (*this)[i] - v[i] );
-	}
-
-	return variant( &res );
+	must_both_be(VARIANT_TYPE::TYPE_LIST, v);
+	return value_cast<game_logic::variant_list>()->list_op(v.value_, [this](variant& v1, variant& v2) { return v1 - v2; });
 }
 
 variant variant::list_elements_mul(const variant& v) const
 {
-	must_be(TYPE_LIST);
-	v.must_be(TYPE_LIST);
-
-	if( num_elements() != v.num_elements() )
-		throw type_error("Operator '.*' requires two lists of the same length");
-
-	std::vector< variant > res;
-	res.reserve(num_elements());
-
-	for(size_t i = 0; i < num_elements(); ++i) {
-		res.push_back( (*this)[i] * v[i] );
-	}
-
-	return variant( &res );
+	must_both_be(VARIANT_TYPE::TYPE_LIST, v);
+	return value_cast<game_logic::variant_list>()->list_op(v.value_, [this](variant& v1, variant& v2) { return v1 * v2; });
 }
 
 variant variant::list_elements_div(const variant& v) const
 {
-	must_be(TYPE_LIST);
-	v.must_be(TYPE_LIST);
-
-	if( num_elements() != v.num_elements() )
-		throw type_error("Operator './' requires two lists of the same length");
-
-	std::vector< variant > res;
-	res.reserve(num_elements());
-
-	for(size_t i = 0; i < num_elements(); ++i) {
-		res.push_back( (*this)[i] / v[i] );
-	}
-
-	return variant( &res );
+	must_both_be(VARIANT_TYPE::TYPE_LIST, v);
+	return value_cast<game_logic::variant_list>()->list_op(v.value_, [this](variant& v1, variant& v2) { return v1 / v2; });
 }
```

Comment:
Hmm, now I'm wondering why there's no `.^` operator – though I guess the use for that would be rather low – or `.%` operator.